src/LuaTransformer.ts

```diff
@@ -3123,12 +3255,13 @@ export class LuaTransformer {
 
         if (classDecorators.has(DecoratorKind.CustomConstructor)) {
             const customDecorator = classDecorators.get(DecoratorKind.CustomConstructor);
-            if (!customDecorator.args[0]) {
+            if (customDecorator === undefined || customDecorator.args[0] === undefined) {
                 throw TSTLErrors.InvalidDecoratorArgumentNumber("@customConstructor", 0, 1, node);
             }
+
             return tstl.createCallExpression(
                 tstl.createIdentifier(customDecorator.args[0]),
-                this.transformArguments(node.arguments),
+                this.transformArguments(node.arguments || ([] as ts.Expression[])),
```

Comment:
Is this `as` necessary? usually TS would infer that sort of thing.

Reply:
Oh I guess I must have done something wrongbefore that it didn't work, you are right it can resolve this.